src/lib/lessons.ts

```diff
@@ -49,38 +77,155 @@ async function loadLessonMetadataFromSanity(slug: string) {
     slug,
   }
 
-  const lesson = await sanityClient.fetch(lessonQuery, params)
+  try {
+    const baseValues = await sanityClient.fetch(lessonQuery, params)
+
+    const derivedValues = deriveDataFromBaseValues(baseValues)
 
-  return lesson
+    return {...baseValues, ...derivedValues}
+  } catch (e) {
+    // Likely a 404 Not Found error
+    console.log('Error fetching from Sanity: ', e)
+
+    return {}
+  }
+}
+
+// TODO: Derive the next_up_url from the collection and lesson slug
```

Comment:
We aren't trying to handle `next_up_url` anymore. The lesson page already knows how to do that.

```suggestion
```